core/store/src/genesis/initialization.rs

```diff
@@ -42,6 +43,17 @@ pub fn initialize_sharded_genesis_state(
         get_genesis_state_roots(&store)
             .expect("Store failed on genesis initialization")
             .expect("Genesis state roots not found in storage");
+        // TODO: with 2.6 release, remove storing genesis height
+        let mut store_update: crate::StoreUpdate = store.store_update();
+        set_genesis_height(&mut store_update, &genesis.config.genesis_height);
+        store_update.commit().expect("Store failed on genesis initialization");
+        let genesis_height = get_genesis_height(&store)
+            .expect("Store failed on genesis initialization")
+            .expect("Genesis height not found in storage");
+        assert_eq!(
+            genesis_height, genesis.config.genesis_height,
+            "Genesis height in store is different from the one in genesis config"
+        );
```

Comment:
This seems like an overkill tbh :)

Reply:
Haha! Yes, I can remove it. Originally wrote it so that this code stays even after we remove the TODO: 2.6 release